neuromation/api/url_utils.py

```diff
@@ -51,7 +51,8 @@ def normalize_storage_path_uri(uri: URL, username: str) -> URL:
     return _normalize_uri(uri, username)
 
 
-def _normalize_uri(uri: URL, username: str) -> URL:
+def _normalize_uri(resource: Union[URL, str], username: str) -> URL:
```

Comment:
Please keep the function untouched but call `_normalize_uri(URL(s), username)` where needed.
Union is not a perfect argument type, here it can be avoided easily.